Add documentation for CPM usage (#1600)
rapidsai/rmm: README.md

`THRUST_HOST_SYSTEM` and `THRUST_DEVICE_SYSTEM`; see
[Thrust's CMake documentation](https://github.com/NVIDIA/cccl/blob/main/thrust/thrust/cmake/README.md).

### Using CPM to manage RMM

> Just like RMM uses [CPM.cmake](https://github.com/cpm-cmake/CPM.cmake)

Reviewer: ```suggestion
RMM uses [CPM.cmake](https://github.com/cpm-cmake/CPM.cmake)
```

> to manage its dependencies on CCCL and others, you can use it for your

Reviewer: ```suggestion
to manage its dependencies on [CCCL](https://github.com/nvidia/cccl) and others, you can use it for your
```

`THRUST_HOST_SYSTEM` and `THRUST_DEVICE_SYSTEM`; see
[Thrust's CMake documentation](https://github.com/NVIDIA/cccl/blob/main/thrust/thrust/cmake/README.md).

### Using CPM to manage RMM

> RMM uses [CPM.cmake](https://github.com/cpm-cmake/CPM.cmake) to manage
> its dependencies on [CCCL](https://github.com/nvidia/cccl) and others,
> you can use it for your project's dependency on RMM itself.

Reviewer 2: ```suggestion
RMM uses [CPM.cmake](https://github.com/cpm-cmake/CPM.cmake) to manage
its dependencies, including [CCCL](https://github.com/nvidia/cccl), and you can
use CPM for your project's dependency on RMM.
```
